tests/conftest.py

```diff
@@ -78,3 +85,114 @@ def generate_token(sig_key: str = ADMIN_SECRET_KEY, payload: JWTPayloadDict | No
             exp=int(time.time()) + ACCESS_TOKEN_EXPIRE_SECONDS,
         )
     return jwt.encode(dict(payload), sig_key, headers=headers)
+
+
+_skip_tables = (
+    'alembic_version',
+    'auth_type',
+    'branding_type',
+    'dm_datetime',
+    'key_types',
+    'notification_status_types',
+    'template_process_type',
+    'provider_details',
+    'provider_details_history',
+    'organisation_types',
+    'invite_status_type',
+    'job_status',
+)
+
+_acceptable_counts = {
+    'communication_items': 4,
+    'job_status': 9,
+    'key_types': 3,
+    # 'provider_details': 9,  # TODO: 1631
+    # 'provider_details_history': 9,  # TODO: 1631
+    'provider_rates': 5,
+    # 'rates': 2,
+    'service_callback_channel': 2,
+    'service_callback_type': 3,
+    'service_permission_types': 12,
+}
+
+
+# No need to cover the test cleanup with a test
+def pytest_sessionfinish(
+    session: pytest.Session,
+    exitstatus: int | pytest.ExitCode,
+) -> None:  # pragma: no cover
+    """Ran after all tests.
+
+    Args:
+        session (Any): The pytest session, not DB related
+        exitstatus (Any): Unknown but mandatory
+    """
+    asyncio.run(_validate_and_clean_tables(session))
+
+
+async def _validate_and_clean_tables(pt_session: pytest.Session) -> None:  # pragma: no cover
+    """Validate and clean tables.
+
+    Args:
+        pt_session (pytest.Session): Pytest Session object, used for setting exit status
+    """
+    from sqlalchemy import select
+
+    from app.db.db_init import get_read_session_with_context, init_db, metadata_legacy
+
+    # Reflect the metadata, prep write session
+    await init_db()
+
+    tables_with_artifacts = []
+    artifact_counts = []
+    print()
+
+    # Use metadata to query the table and add the table name to the list if there are any records
+    async with get_read_session_with_context() as session:
+        for _, v in metadata_legacy.tables.items():
```

Comment:
Since you're not using the keys, are you able to call "values" rather than "items" and omit the underscore?

Reply:
Old habits, nice catch.